cpp/memilio/io/result_io.cpp

```diff
@@ -89,6 +89,118 @@ IOResult<void> save_result(const std::vector<TimeSeries<double>>& results, const
+        const int num_timepoints          = static_cast<int>(result.get_num_time_points());
+        constexpr int num_infectionstates = 3; // (int)result.get_num_elements() / num_groups;
+
+        hsize_t dims_t[] = {static_cast<hsize_t>(num_timepoints)};
+        H5DataSpace dspace_t{H5Screate_simple(1, dims_t, NULL)};
+        MEMILIO_H5_CHECK(dspace_t.id, StatusCode::UnknownError, "Time DataSpace could not be created.");
+        H5DataSet dset_t{H5Dcreate(start_node_h5group.id, "Time", H5T_NATIVE_DOUBLE, dspace_t.id, H5P_DEFAULT,
+                                   H5P_DEFAULT, H5P_DEFAULT)};
+        MEMILIO_H5_CHECK(dset_t.id, StatusCode::UnknownError, "Time DataSet could not be created (Time).");
+        auto values_t = std::vector<double>(result.get_times().begin(), result.get_times().end());
+        MEMILIO_H5_CHECK(H5Dwrite(dset_t.id, H5T_NATIVE_DOUBLE, H5S_ALL, H5S_ALL, H5P_DEFAULT, values_t.data()),
+                         StatusCode::UnknownError, "Time data could not be written.");
+
+        int start_id = ids[edge_indx].first;
+        auto total   = Eigen::Matrix<double, Eigen::Dynamic, Eigen::Dynamic, Eigen::RowMajor>::Zero(num_timepoints,
+                                                                                                  num_infectionstates)
+                         .eval();
+        while (edge_indx < num_edges && ids[edge_indx].first == start_id) {
+            const auto& result_edge = results[edge_indx];
+            auto edge_result        = Eigen::Matrix<double, Eigen::Dynamic, Eigen::Dynamic, Eigen::RowMajor>::Zero(
+                                   num_timepoints, num_infectionstates)
+                                   .eval();
+            for (Eigen::Index t_idx = 0; t_idx < result_edge.get_num_time_points(); ++t_idx) {
+                auto v                 = result_edge.get_value(t_idx).transpose().eval();
+                edge_result.row(t_idx) = v;
+                total.row(t_idx) += v;
+            }
+
+            hsize_t dims_values[] = {static_cast<hsize_t>(num_timepoints), static_cast<hsize_t>(num_infectionstates)};
+            H5DataSpace dspace_values{H5Screate_simple(2, dims_values, NULL)};
+            MEMILIO_H5_CHECK(dspace_values.id, StatusCode::UnknownError, "Values DataSpace could not be created.");
+            auto dset_name = "End" + std::to_string(ids[edge_indx].second);
+            H5DataSet dset_values{H5Dcreate(start_node_h5group.id, dset_name.c_str(), H5T_NATIVE_DOUBLE,
+                                            dspace_values.id, H5P_DEFAULT, H5P_DEFAULT, H5P_DEFAULT)};
+            MEMILIO_H5_CHECK(dset_values.id, StatusCode::UnknownError, "Values DataSet could not be created.");
+
+            MEMILIO_H5_CHECK(
+                H5Dwrite(dset_values.id, H5T_NATIVE_DOUBLE, H5S_ALL, H5S_ALL, H5P_DEFAULT, edge_result.data()),
+                StatusCode::UnknownError, "Values data could not be written.");
```

Comment:
The error messages in this function are not very helpfull, if you don't already know what you are looking for. Can you expand these?